src/Avalonia.Base/Media/Brush.cs

```diff
@@ -90,14 +90,26 @@ public static IBrush Parse(string s)
         
         protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs change)
         {
-            if (change.Property == TransformProperty) 
+            if (change.Property == TransformProperty)
+            {
+                var (oldValue, newValue) = change.GetOldAndNewValue<ITransform>();
+
+                if (oldValue is IMutableTransform oldMutableValue)
+                    oldMutableValue.Changed -= OnTransformChanged;
+                
+                if (newValue is IMutableTransform newMutableValue)
+                    newMutableValue.Changed += OnTransformChanged;
                 _resource.ProcessPropertyChangeNotification(change);
```

Comment:
This method is supposed to properly handle the change notification including nested resource changes, so the fix is incorrect _and_ will cause memory leaks if transform is reused

Reply:
Is there a client method to notify server if the resources change and need to be reserialized?

Reply:
Transform itself is supposed to notify the server about its changes. The bug is most likely on the server-side.